lib/Yield/HexDump.cpp

```diff
@@ -0,0 +1,280 @@
+                           inplace_plus,
+                           inter_section,
+                           discrete_interval<MetaAddress>>;
+  Map Instructions;
+  ptml::PTMLBuilder PTMLBuilder;
+
+  auto CreateTag = [&PTMLBuilder](const std::string &Location) -> AddressTag {
+    auto Tag = PTMLBuilder.getTag("span");
+    Tag.addAttribute("data-location-definition", Location);
+    return { Location, Tag };
+  };
+
+  for (const Function &F :
+       FunctionTags::Isolated.functions(&Module.getModule())) {
+    const efa::FunctionMetadata &Metadata = FunctionMetadataCache
+                                              .getFunctionMetadata(&F);
+    const MetaAddress EntryAddress = Metadata.Entry();
+
+    for (const BasicBlock &BB : F) {
+      for (const Instruction &I : BB) {
+        if (auto *Call = getCallTo(&I, "newpc")) {
+          const BasicBlock *JumpTarget = getJumpTargetBlock(&BB);
+          revng_assert(JumpTarget != nullptr);
+          const BasicBlockID
+            BasicBlockID = blockIDFromNewPC(JumpTarget->getFirstNonPHI());
+
+          const MetaAddress
+            Address = MetaAddress::fromValue(Call->getArgOperand(0));
+
+          const auto *SizeValue = dyn_cast<ConstantInt>(Call->getArgOperand(1));
+
+          uint64_t Size = SizeValue->getZExtValue();
+          MetaAddress Begin = Address.toGeneric();
+          MetaAddress End = Begin + Size;
+          auto Interval = discrete_interval<MetaAddress>::right_open(Begin,
+                                                                     End);
+
+          const std::string Str = serializedLocation(ranks::Instruction,
+                                                     EntryAddress,
+                                                     BasicBlockID,
+                                                     Address);
+
+          const std::set<AddressTag> Set{ CreateTag(Str) };
+          Instructions.add(std::make_pair(Interval, Set));
+        }
+      }
+    }
+  }
+
+  Output << "<div>\n";
+
+  auto FormatNumber = [](uint64_t Number, unsigned Width = 8) {
+    return FormattedNumber(Number, 0, Width, true, false, false);
+  };
+
+  MetaAddress CurrentAddress;
+  Map::const_iterator Current = Instructions.begin();
+  Map::const_iterator Next = std::next(Current);
+  const Map::const_iterator End = Instructions.end();
+
+  for (const model::Segment &Segment : Binary->Segments()) {
+    uint64_t Offset = Segment.StartOffset();
+    MetaAddress Address = Segment.StartAddress();
+    uint64_t Size = Segment.VirtualSize();
+
+    CurrentAddress = Address;
+
+    std::optional<llvm::ArrayRef<uint8_t>>
+      SegmentBinary = BinaryView.getByAddress(Address, Size);
+
+    const size_t BytesInLine = 16;
+
+    size_t Counter = 0;
+
+    std::set<ptml::Tag> PreviousAddresses;
+    bool SpanIsOpen = false;
+    if (SegmentBinary.has_value()) {
+      SmallString<16> PrintableChars;
+      for (size_t Index = 0; Index < SegmentBinary->size(); ++Index) {
```

Comment:
Please thoroughly comment all of this loop, it's rather delicate and not obvious to follow.